api/envoy/api/v2/core/protocol.proto

```diff
@@ -35,6 +35,20 @@ message HttpProtocolOptions {
 }
 
 message Http1ProtocolOptions {
+  message HeaderKeyFormat {
+    message ProperCaseWords {
+    }
+
+    oneof header_format {
+      option (validate.required) = true;
+
+      // Formats the header by proper casing words: the first character and any character following
+      // a special character will be capitalized if it's an alpha character. For example,
+      // "content-type" becomes "Content-Type", and "foo$b#$are" becomes "Foo$B#$Are".
+      ProperCaseWords proper_case_words = 1;
```

Comment:
Optional: Do you think it's worth calling out both that this is an effort to normalize HTTP/1 headers to what users expect, and that this will not behave "as expected" for all headers, e.g. TE?  I can imagine that folks who have HTTP/1 casing issues will think this is the solution to their problem and I think documenting the simplicity vs the non-handled corner cases will probably save someone some trouble.